app/assets/javascripts/components/_all_skills.js.jsx

```diff
@@ -0,0 +1,29 @@
+var AllSkills = React.createClass({
+  handleDelete(id) {
+    this.props.handleDelete(id);
+  },
+
+  handleEdit() {
+    var id      = this.props.skill.id;
+    var name    = this.refs.name.value;
+    var details = this.refs.details.value;
+    var level   = this.props.skill.level;
+
+    var skill = {id: id, name: name, details: details, level: level }
+  },
+
+  handleUpdate(skill) {
+  console.log(skill, 'in handleUpdate');
+},
+
+render() {
+  return (
+    <div>
+      <NewSkill handleSubmit={this.handleSubmit} />
+      <AllSkills skills={this.state.skills}
```

Comment:
This might be what is messing things up. In the `AllSkills` component you are rendering the `AllSkills` component. I think you should iterate over `this.props.skills` and for each skill create a `Skill` component, and render them all in the AllSkills component.